### ISSUE TITLE
GROUP BY time returns inconsistent results with each query.

### ISSUE THREAD

**Issue by lpalm** [influxdata/influxdb#6996]
I have seen this behavior under v1.0.0-beta1 (Fedora 22).

While data is being inserted, entire counts for certain hours will disappear and sometimes re-appear. Values are also pretty unreliable between queries. The latter issue is easily reproducible. See below.

My query:

`select count(someField) from someMeasurement where time > now() - 18h group by time(1h)`

Here are the values for the "2016-07-11T13:00:00Z" bucket (7 hours prior to query time) for successive queries. I'm just hitting enter repeatedly with the query above on the web admin page, a few seconds apart.

```
805933
820059 // Increases
811868 // Decreases
811876
811819
812328
812472
811868 // Some values are repeated from previous queries
820059 // Some values are repeated from previous queries
812540
```

Notice that the amount both increases and decreases. We're not deleting any points from the database, only inserting (both backfilling >10 days ago, which shouldn't appear in this window, and real-time insertion with <5s delay). I could understand if the counts only increased (maybe some straggler points just got added), but not this random fluctuation.

Let me know if I can provide additional information

**Comment by lpalm** [comment]
In case the returned buckets are not aligned at the hour boundary (they should, if not we should return a correct timestamp?), here's the results using a fixed time range.

select count(someField) from someMeasurement where time >= '2016-07-11T00:00:00Z' AND time <= '2016-07-11T18:00:00Z' group by time(1h)

Results for successive queries for the 2016-07-11T13:00:00Z window:
306722
306717
306717
295477
298166
304477

Also notice how those numbers are very different to the ones posted on the original issue request, despite the query being over the same measurement, field and time bucket. Over half of the points vanished (800k -> 300k).

**Comment by BrannonKing** [comment]
Has anything been added to the later betas that might address this issue?

**Comment by jsternberg** [comment]
I think this might be fixed in beta3. Did you reinsert any data after the initial insert? beta1 and beta2 contain a bug where it will return data out of order sometimes when you do that.

**Comment by lpalm** [comment]
Yes, it is quite likely that some data was reinserted. I'll keep an eye once and update this issue once we're on the latest beta.

**Comment by jwilder** [comment]
This should be fixed as of 1.0.0 or later via #7038

**Comment by sameer2800** [comment]
hey all, I am seeing this issue in 1.8.3version.
